src/backend/catalog/namespace.c

```diff
@@ -4554,31 +4548,16 @@ RemoveTempRelationsCallback(int code, Datum arg)
 		return;
 	}
 
+	/* If there is a temporary namespace, schedule its deletion */
 	if (OidIsValid(myTempNamespace))
+		GpScheduleSessionReset();
```

Comment:
How are `Temp Namespace` and `Session Reset` related?

Reply:
`GpScheduleSessionReset` also registers the temporary namespace for deletion (moves it from `myTempNamespace` to `OldTempNamespace` global variable). `TempNamespace` functions then interact with `OldTempNamespace`.